targets/wasm-unknown.json

```diff
@@ -0,0 +1,26 @@
+{
+	"llvm-target":   "wasm32-unknown-unknown",
+	"cpu":           "generic",
+	"features":      "+mutable-globals,+nontrapping-fptoint,+sign-ext",
+	"build-tags":    ["tinygo.wasm", "wasm_unknown"],
+	"goos":          "linux",
+	"goarch":        "arm",
+	"linker":        "wasm-ld",
+	"rtlib":         "compiler-rt",
+	"scheduler":     "none",
+	"default-stack-size": 4096,
+	"cflags": [
+		"-mno-bulk-memory",
```

Comment:
Note that wasm-ld by default _does_ include bulk memory instructions, which will likely be invoked by the compiler sometimes (for example, when copying a large slice).

Reply:
But this target shouldn't be using wasi-libc anyways since there *is* no `wasi` here.

Reply:
Oh, I thought the memory ones were in the compiler runtime lib.  Totally forgot about the math ones though.  I was thinking about the file access bits and spaced on everything else.

Reply:
There has been discussion inside LLVM of adding the math functions to compiler-rt which would in fact fix some subtle bugs in LLVM (the libc versions may set errno while the LLVM intrinsics don't, so if the LLVM intrinsics get lowered to the libc versions they might unexpectedly set errno). But I haven't seen anybody actually put in the work to make it happen - probably because this is only relevant when compiling with `-fno-math-errno` and targeting instruction sets that don't have these math functions as intrinsics in the instruction set. In other words, nobody really cares enough.

Regarding the memory builtins: I haven't seen any discussion on those, I think they're in libc for historic reasons. I suspect that building with bulk memory instructions enabled will avoid those intrinsics - but then the host needs support for those functions.

What we _could_ do is make a new "minimal wasm libc" based on wasi-libc but keeping only the parts we need (like memory and math functions) and excluding any WASI parts. I don't even think this will be very difficult: just compile a subset of the files in wasi-libc in a similar way that libraries like compiler-rt and picolibc are built today (on demand and cached).